src/google-maps/map-rectangle/map-rectangle.ts

```diff
@@ -0,0 +1,223 @@
+
+  /**
+   * @see
+   * developers.google.com/maps/documentation/javascript/reference/polygon#Rectangle.dblclick
+   */
+  @Output()
+  rectangleDblclick: Observable<google.maps.MouseEvent> =
+      this._eventManager.getLazyEmitter<google.maps.MouseEvent>('dblclick');
+
+  /**
+   * @see
+   * developers.google.com/maps/documentation/javascript/reference/polygon#Rectangle.drag
+   */
+  @Output()
+  rectangleDrag: Observable<google.maps.MouseEvent> =
+      this._eventManager.getLazyEmitter<google.maps.MouseEvent>('drag');
+
+  /**
+   * @see
+   * developers.google.com/maps/documentation/javascript/reference/polygon#Rectangle.dragend
+   */
+  @Output()
+  rectangleDragend: Observable<google.maps.MouseEvent> =
+      this._eventManager.getLazyEmitter<google.maps.MouseEvent>('dragend');
+
+  /**
+   * @see
+   * developers.google.com/maps/documentation/javascript/reference/polygon#Rectangle.dragstart
+   */
+  @Output()
+  rectangleDragstart: Observable<google.maps.MouseEvent> =
+      this._eventManager.getLazyEmitter<google.maps.MouseEvent>('dragstart');
+
+  /**
+   * @see
+   * developers.google.com/maps/documentation/javascript/reference/polygon#Rectangle.mousedown
+   */
+  @Output()
+  rectangleMousedown: Observable<google.maps.MouseEvent> =
+      this._eventManager.getLazyEmitter<google.maps.MouseEvent>('mousedown');
+
+  /**
+   * @see
+   * developers.google.com/maps/documentation/javascript/reference/polygon#Rectangle.mousemove
+   */
+  @Output()
+  rectangleMousemove: Observable<google.maps.MouseEvent> =
+      this._eventManager.getLazyEmitter<google.maps.MouseEvent>('mousemove');
+
+  /**
+   * @see
+   * developers.google.com/maps/documentation/javascript/reference/polygon#Rectangle.mouseout
+   */
+  @Output()
+  rectangleMouseout: Observable<google.maps.MouseEvent> =
+      this._eventManager.getLazyEmitter<google.maps.MouseEvent>('mouseout');
+
+  /**
+   * @see
+   * developers.google.com/maps/documentation/javascript/reference/polygon#Rectangle.mouseover
+   */
+  @Output()
+  rectangleMouseover: Observable<google.maps.MouseEvent> =
+      this._eventManager.getLazyEmitter<google.maps.MouseEvent>('mouseover');
+
+  /**
+   * @see
+   * developers.google.com/maps/documentation/javascript/reference/polygon#Rectangle.mouseup
+   */
+  @Output()
+  rectangleMouseup: Observable<google.maps.MouseEvent> =
+      this._eventManager.getLazyEmitter<google.maps.MouseEvent>('mouseup');
+
+  /**
+   * @see
+   * developers.google.com/maps/documentation/javascript/reference/polygon#Rectangle.rightclick
+   */
+  @Output()
+  rectangleRightclick: Observable<google.maps.MouseEvent> =
+      this._eventManager.getLazyEmitter<google.maps.MouseEvent>('rightclick');
+
+  constructor(private readonly _map: GoogleMap) {}
+
+  ngOnInit() {
+    const combinedOptionsChanges = this._combineOptions();
+
+    combinedOptionsChanges.pipe(take(1)).subscribe(options => {
+      this._rectangle = new google.maps.Rectangle(options);
```

Comment:
I see. Does fixing this rely on the NgZone change or should I fix this now?

Reply:
I think that it relies on the `NgZone` changes since they update the event manager to emit events in the Zone. If this is urgent we can merge it in and I'll do a follow up PR to apply the same fix to the rectangle.

Reply:
Okay, no rush on rectangle though. It's not going into any release yet, it's the separate "google-maps-next" branch.